beacon-chain/casper/incentives.go

```diff
@@ -100,16 +100,18 @@ func QuadraticPenalty(numberOfSlots uint64) uint64 {
 
 // RewardValidatorCrosslink applies rewards to validators part of a shard committee for voting on a shard.
 // TODO(#538): Change this to big.Int as tests using 64 bit integers fail due to integer overflow.
-func RewardValidatorCrosslink(totalDeposit uint64, participatedDeposits uint64, rewardQuotient uint64, validator *pb.ValidatorRecord) {
-	currentBalance := int64(validator.Balance)
-	currentBalance += int64(currentBalance) / int64(rewardQuotient) * (2*int64(participatedDeposits) - int64(totalDeposit)) / int64(totalDeposit)
-	validator.Balance = uint64(currentBalance)
+func RewardValidatorCrosslink(totalDeposit uint64, participatedDeposits uint64, rewardQuotient uint64, validator *pb.ValidatorRecord) (newValidator *pb.ValidatorRecord) {
+	newValidator = validator
```

Comment:
Same over here, you are referencing the same validator record with `newValidator`